### ISSUE TITLE
Potentially inconsistent wordsplitting behaviour on export assignment

### ISSUE THREAD

**Issue by Earnestly** [emersion/mrsh#150]
According to [wooledge](https://mywiki.wooledge.org/BashPitfalls#local_var.3D.24.28cmd.29) it is possible that a shell may induce wordsplitting on export assignment if the expansion isn't properly quoted (the example is for local but mentions the case is the same for export).  The wiki claims dash is susceptible to this but I did not encounter the behaviour:

```sh
#!/bin/sh --
var='contains spaces'
export ENVAR=$var
printf '[%s]\n' "$ENVAR"
```

```
% for shell in bash dash sh zsh mksh mrsh; do printf '%s: ' "$shell"; "$shell" foo.sh; done
bash: [contains spaces]
dash: [contains spaces]
sh: [contains spaces]
zsh: [contains spaces]
mksh: [contains spaces]
mrsh: [contains]
```

Of the shells I have, mrsh is the only one which exhibits the behaviour wooledge suggests is possible but I'm unsure if this is a bug or merely a different interpretation of POSIX.

**Comment by emersion** [comment]
On dash:

```console
> dash
$ v='a b c'
$ export a=$v
$ echo $a
a
```

Are you sure you're running the script with the correct shell? (ie. `dash script.sh` instead of running `./script.sh` in `dash`)

**Comment by Earnestly** [comment]
That's right.  You should be able to see the exact command line I used in the original report which calls `"$shell" foo.sh` where `foo.sh` is the script I used.

What you're seeing is also something `shbot` on freenode.net/#bash gets when using the `#sh` trigger (which uses dash) but perhaps there is a version difference?  My dash is currently 0.5.10.2

**Comment by Earnestly** [comment]
Indeed this is a recent change in dash, see: https://git.kernel.org/pub/scm/utils/dash/dash.git/commit/?id=cbb71a836874d176809a34e22f6b6e4e3ba8c85b

**Comment by emersion** [comment]
Makes sense. It seems like mrsh's behavior is POSIX-compliant at least for now. I'll leave this open just to remember about all of this (the dash commit says it'll become POSIX "soon").

**Comment by illiliti** [comment]
on [p]bosh, yash

```
zsh >> bosh test.sh
[contains]
zsh >> pbosh test.sh
[contains]
zsh >> yash test.sh
[contains]
zsh >>
```

**Comment by Earnestly** [comment]
Has this issue been affected by any changes recently?  I now seem to not be having the same splitting behaviour:

```sh
var='with spaces'
export env=$var
printf '[%s]\n' "$var"
# => [with spaces]
```
